kore/src/Kore/Internal/SideCondition.hs

```diff
@@ -0,0 +1,123 @@
+{-|
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+-}
+
+module Kore.Internal.SideCondition
+    ( SideCondition  -- Constructor not exported on purpose
+    , addAssumedTrue
+    , assumeTrueCondition
+    , assumeTruePredicate
+    , mapVariables
+    , top
+    , topTODO
+    , toPredicate
+    ) where
+
+import Kore.Attribute.Pattern.FreeVariables
+    ( HasFreeVariables (..)
+    )
+import Kore.Internal.Condition
+    ( Condition
+    )
+import qualified Kore.Internal.Condition as Condition
+    ( andCondition
+    , fromPredicate
+    , mapVariables
+    , toPredicate
+    , top
+    )
+import Kore.Internal.Predicate
+    ( Predicate
+    )
+import Kore.Internal.Variable
+    ( InternalVariable
+    )
+import Kore.TopBottom
+    ( TopBottom (..)
+    )
+import Kore.Unparser
+    ( Unparse (..)
+    )
+
+newtype SideCondition variable =
```

Comment:
Could you add a comment here to indicate something like: this is a `Condition` used in the evaluation context which is not added to the result?